### PR TITLE
Test Added for http, Module, process, Timer

### PR DESCRIPTION
**Pull request by chunseoklee**
1. request, connection, close event for http.Server
2. response, socket event for http.ClientRequest
3. Restore require test
4. Test added for process.nextTick
5. Test added for Timers.clearTimeout

IoT.js-DCO-1.0-Signed-off-by: Chunseok Lee chunseok.lee@samsung.com

### CODE REVIEW

**Comment by wateret** [comment]
LGTM

File: test/run_pass/require_add.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+/*
+  @NOTEST
+*/
+
+var x = require("buffer");
+
+exports.add = function(a, b) {
+  return a + b;
+};
```
Comment (ILyoan) [inline test/run_pass/require_add.js]: I think this test set is already there in 'test/run_pass/require1`

Comment (chunseoklee) [inline test/run_pass/require_add.js]: @ILyoan Yes, I just copy it. It there any way to add a test in subdirectory into our test set ?

Comment (ILyoan) [inline test/run_pass/require_add.js]: @chunseoklee "check_test.py" will find every javascript file under 'run_pass' and 'run_fail' directories including subdirectories except those files with `@NOTEST` annotation.

**Comment by chunseoklee** [comment]
@ILyoan test_require.js removed from run_pass.
And add NOTEST into require1/module_cache.js

**Comment by ILyoan** [comment]
LGTM

**Comment by ILyoan** [comment]
Related to #238

**Pull request merged**